src/parser.c

```diff
@@ -494,53 +584,68 @@ struct l_queue *parser_queue_from_json_array(json_object *jobj,
 
 	for (i = 0; i < len; i++) {
 		jobjentry = json_object_array_get_idx(jarray, i);
-
-		item = foreach_cb(jobjentry);
+		item = device_array_item(jobjentry, item_cb);
 		if (item)
 			l_queue_push_tail(list, item);
 	}
-
+	json_object_put(jobj);
 	return list;
 }
 
-struct l_queue *parser_request_to_list(json_object *jso)
+struct l_queue *parser_request_to_list(const char *json_str)
 {
+	json_object *jso;
 	struct l_queue *list;
 	json_object *json_array;
 	json_object *jobjentry;
-	int sensor_id;
+	int sensor_id;	
 	uint64_t i;
+	bool has_err;
 
+	jso = json_tokener_parse(json_str);
+	if (!jso)
+		return NULL;
 	list = l_queue_new();
 
-	if (!json_object_object_get_ex(jso, "sensorIds", &json_array))
-		goto fail;
+	if (!json_object_object_get_ex(jso, KNOT_JSON_FIELD_SENSOR_IDS,
+			&json_array)) {
+		l_queue_destroy(list, l_free);
+		return NULL;
+	}
 
+	has_err = false;			
```

Comment:
Please, remove these tabs.
```suggestion
	has_err = false;
```